csharp/device/Microsoft.Azure.Devices.Client/IotHubConnectionStringBuilder.cs

```diff
@@ -165,10 +215,37 @@ void Parse(string iotHubConnectionString)
             this.SharedAccessSignature = GetConnectionStringOptionalValue(map, SharedAccessSignaturePropertyName);
 
             this.Validate();
+#endif
         }
 
         void Validate()
         {
+#if NETMF
+            if (this.DeviceId.IsNullOrWhiteSpace())
+            {
+                throw new ArgumentException("DeviceId must be specified in connection string");
+            }
+
+            if (!(this.SharedAccessKey.IsNullOrWhiteSpace() ^ this.SharedAccessSignature.IsNullOrWhiteSpace()))
+            {
+                throw new ArgumentException("Should specify either SharedAccessKey or SharedAccessSignature");
+            }
+
+            if (this.IotHubName.IsNullOrWhiteSpace())
+            {
+                throw new ArgumentException("Missing IOT hub name");
+            }
+
+            if (!this.SharedAccessKey.IsNullOrWhiteSpace())
+            {
+                Convert.FromBase64String(this.SharedAccessKey);
+            }
+
+            //if (SharedAccessSignatureParser.IsSharedAccessSignature(this.SharedAccessSignature))
```

Comment:
Probably yes. I choose not to implement that validation. Not only it's costly to do but it's not necessary. For NETMF implementations the signature is computed in the code so it's doesn't make much sense to me checking something that you've just computed.